ios/brave-ios/Sources/Brave/Frontend/Browser/New Tab Page/Backgrounds/NewTabPageVideoButtonsView.swift

```diff
@@ -0,0 +1,194 @@
+      return
+    }
+
+    guard let playStarted = tappedVideoBackground?() else {
+      return
+    }
+
+    if playStarted {
+      playPauseButtonImage.setImage(imageName: "leo.play.circle")
+      showAndFadeOutImage(imageView: playPauseButtonImage)
+    } else {
+      playPauseButtonImage.setImage(imageName: "leo.pause.circle")
+      showAndFadeOutImage(imageView: playPauseButtonImage)
+    }
+  }
+
+  private func showAndFadeOutImage(imageView: UIView) {
+    imageView.alpha = 0
+    UIView.animate(
+      withDuration: 0.1,
+      animations: {
+        imageView.alpha = 1
+      },
+      completion: { _ in
+        UIView.animate(
+          withDuration: 0.3,
+          delay: 0.5,
+          animations: {
+            imageView.alpha = 0
+          }
+        )
+      }
+    )
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+}
+
+extension NewTabPageVideoButtonsView {
+  private class CancelButton: SpringButton {
+    let imageView = UIImageView(
+      image: UIImage(braveSystemNamed: "leo.close", compatibleWith: nil)!
+    ).then {
+      $0.tintColor = .white
+      $0.contentMode = .scaleAspectFit
+    }
+
+    private let backgroundView = UIVisualEffectView(
+      effect: UIBlurEffect(style: .systemThinMaterialDark)
+    ).then {
+      $0.clipsToBounds = true
+      $0.isUserInteractionEnabled = false
+    }
+
+    override init(frame: CGRect) {
+      super.init(frame: frame)
+
+      clipsToBounds = true
+
+      addSubview(backgroundView)
+
+      backgroundView.contentView.addSubview(imageView)
+
+      backgroundView.snp.makeConstraints {
+        $0.edges.equalToSuperview()
+      }
+
+      imageView.snp.makeConstraints {
+        $0.edges.equalToSuperview().inset(UIEdgeInsets(equalInset: 4))
+      }
+    }
+
+    override func layoutSubviews() {
+      super.layoutSubviews()
+
+      layer.cornerRadius = bounds.height / 2.0
+    }
+  }
+
+  private class BluredImageView: UIView {
+    private let backgroundView = UIVisualEffectView(
+      effect: UIBlurEffect(style: .systemUltraThinMaterialDark)
+    ).then {
+      $0.clipsToBounds = true
+      $0.isUserInteractionEnabled = false
+    }
+    private var imageView = UIImageView().then {
+      $0.tintColor = .white
+      $0.contentMode = .scaleAspectFit
+    }
+
+    init() {
+      super.init(frame: .zero)
+
+      clipsToBounds = true
+
+      addSubview(backgroundView)
+
+      backgroundView.contentView.addSubview(imageView)
+
+      backgroundView.snp.makeConstraints {
+        $0.edges.equalToSuperview()
+      }
+
+      imageView.snp.makeConstraints {
+        $0.edges.equalToSuperview().inset(UIEdgeInsets(equalInset: 10))
+      }
+    }
+
+    override func layoutSubviews() {
+      super.layoutSubviews()
+      layer.cornerRadius = bounds.height / 2.0
+    }
+
+    func setImage(imageName: String) {
+      imageView.image = UIImage(braveSystemNamed: imageName, compatibleWith: nil)!
+    }
```

Comment:
Lets just remove this, `imageView` is already accessible from outside and its better to be explicit that this is taking Nala/Leo names